client/cli/src/lib.rs

```diff
@@ -931,6 +931,28 @@ where
 	Ok(config)
 }
 
+fn interface_str(
+	is_external: bool,
+	is_unsafe_external: bool,
+	is_validator: bool
+) -> Result<&'static str, error::Error> {
+	if is_external && is_validator {
+		return Err(error::Error::Input("--rpc-external and --ws-external options shouldn't be \
+		used if the node is running as vaidator.".to_owned()));
+	}
+
+	if is_external {
+		log::warn!("It isn't safe to expose RPC publicly without a proxy server. \
+		Use `--unsafe-rpc-external` to suppress the warning if you understand the risks.");
```

Comment:
To be fair I wouldn't give an option to suppress the warning, I think `unsafe` should just be used to be able to run the node even if you are a validator. I'd keep the warning whenever you run with `is_external || is_unsafe_external`.

If you agree we should move this message to the validator error instead and update the condition above.